scala-lightautoml-transformers/src/main/scala/org/apache/spark/ml/feature/lightautoml/IsHolidayTransformer.scala

```diff
@@ -0,0 +1,162 @@
+    val func = udf((col_name: String, date: String) => {
+      holidaysDatesBcst.value(col_name)(date)
+    })
+
+    val dt_format = "yyyy-MM-dd"
+    val outColumns = getInputCols.zip(getOutputCols).map {
+      case (in_col, out_col) =>
+        val dt_col = dataset.schema(in_col).dataType match {
+          case _: DateType => date_format(to_timestamp(col(in_col)), dt_format)
+          case _: TimestampType => date_format(col(in_col), dt_format)
+          case _: LongType => date_format(col(in_col).cast(TimestampType), dt_format)
+          case _: IntegerType => date_format(col(in_col).cast(TimestampType), dt_format)
+          case _: StringType => col(in_col)
+        }
+        func(lit(in_col).cast(StringType), dt_col).alias(out_col)
+    }
+
+    dataset.withColumns(getOutputCols, outColumns)
+  }
+
+  override def copy(extra: ParamMap): Transformer = {
+    val copied = new IsHolidayTransformer(uid, holidays_dates)
+    copyValues(copied, extra)
+  }
+
+  override def write: IsHolidayTransformerWriter = new IsHolidayTransformerWriter(this)
+
+  override def toString: String = {
+    s"IsHolidayTransformer: uid=$uid, " +
+            get(inputCols).map(c => s", numInputCols=${c.length}").getOrElse("") +
+            get(outputCols).map(c => s", numOutputCols=${c.length}").getOrElse("")
+  }
+}
+
+
+object IsHolidayTransformer extends MLReadable[IsHolidayTransformer] {
+  override def read: MLReader[IsHolidayTransformer] = new IsHolidayTransformerReader
+
+  override def load(path: String): IsHolidayTransformer = super.load(path)
+
+  private[IsHolidayTransformer] class IsHolidayTransformerWriter(instance: IsHolidayTransformer)
+          extends MLWriter {
+
+    private case class Data(holidays_dates: Map[String, Array[String]])
+
+    override protected def saveImpl(path: String): Unit = {
+      DefaultParamsWriter.saveMetadata(instance, path, sc)
+      val data = Data(instance.getHolidaysDates.get.map{case (col_name, dates) => (col_name, dates.toArray)}.toMap)
+      val dataPath = new Path(path, "data").toString
+      sparkSession.createDataFrame(Seq(data)).repartition(1).write.parquet(dataPath)
+    }
+  }
+
+  private[IsHolidayTransformer] class IsHolidayTransformerReader extends MLReader[IsHolidayTransformer] {
+
+    private val className = classOf[IsHolidayTransformer].getName
+
+    override def load(path: String): IsHolidayTransformer = {
+      val metadata = DefaultParamsReader.loadMetadata(path, sc, className)
+      val dataPath = new Path(path, "data").toString
+
+      // We support loading old `StringIndexerModel` saved by previous Spark versions.
+      // Previous model has `labels`, but new model has `labelsArray`.
+      val (majorVersion, _) = majorMinorVersion(metadata.sparkVersion)
+      if (majorVersion < 3) {
+        throw new RuntimeException("Spark version < 3 is not supported")
+      }
+
+      val data = sparkSession.read.parquet(dataPath)
```

Comment:
we are not going to support other formats (e.g orc), aren't we?

Reply:
No, it is intended to be only accessible by reader/writer for the transformer, not directly by user.
That's why we opted to choose a specific format. It is not a subject for change from the user's perspective.